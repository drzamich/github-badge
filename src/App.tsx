import React from 'react';
import styled from 'styled-components';
import { Badge } from './Badge';
import { ErrorBox } from './ErrorBox';
import { Loading } from './Loading';
import { Search } from './Search';
import { UserApiResponse } from './interfaces';
import { getResponse } from './apiHandler';

const Container = styled.div`
`;

const App: React.FC = () => {
  const [userApiResponse, setUserApiReponse] = React.useState<UserApiResponse | null>(null);

  const onSearch = async (username: string) => {
    const apiReponse = await getResponse(username);
    setUserApiReponse(apiReponse);
  };

  return (
    <>
      <Search onSearch={onSearch} />
      <Container>
        <Loading />
        {userApiResponse?.user
            && <Badge user={userApiResponse.user} />}
        {userApiResponse?.error
            && <ErrorBox error={userApiResponse?.error} />}
      </Container>
    </>
  );
};

export default App;
