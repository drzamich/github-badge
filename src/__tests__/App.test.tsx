import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

test('renders the app without crashing and matches snapshot', () => {
  const app = render(<App />);
  expect(app).toMatchSnapshot();
});
