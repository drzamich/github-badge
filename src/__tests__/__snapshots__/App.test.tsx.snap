// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app without crashing and matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-pFZIQ jgyxMO"
      >
        <input
          autocomplete="off"
          class="sc-jrAGrp bAGLIR"
          placeholder="Search for users"
          type="text"
          value=""
        />
        <button
          class="sc-kEjbxe gscWDP"
          type="button"
        >
          Search
        </button>
      </div>
      <div
        class="sc-iqHYGH IbLJK"
      >
        <h3
          class="sc-bdfBwQ fVVSDW"
        >
          Use the search input to look for a user.
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-pFZIQ jgyxMO"
    >
      <input
        autocomplete="off"
        class="sc-jrAGrp bAGLIR"
        placeholder="Search for users"
        type="text"
        value=""
      />
      <button
        class="sc-kEjbxe gscWDP"
        type="button"
      >
        Search
      </button>
    </div>
    <div
      class="sc-iqHYGH IbLJK"
    >
      <h3
        class="sc-bdfBwQ fVVSDW"
      >
        Use the search input to look for a user.
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
